### DISCUSSION TITLE
Using the training dataset to test the model.

### DISCUSSION THREAD

**Discussion by raminsalmas** [sebp/scikit-survival#282] [General]
I'm wondering why the training dataset (data_x_numeric) is used to test the model in code below:
It doesn't make sense to me.
 https://scikit-survival.readthedocs.io/en/stable/user_guide/00-introduction.html

prediction = estimator.predict(data_x_numeric)
result = concordance_index_censored(data_y["Status"], data_y["Survival_in_days"], prediction)

c_index = 0.7362562471603816

**Comment by sebp** [comment]
You are correct that you should always evaluate on an independent test set, which is done in the [user guide on evaluation](https://scikit-survival.readthedocs.io/en/stable/user_guide/evaluating-survival-models.html).

This particular example in the introduction guide is just to illustrate the basic API of sksurv, not best practices.

**Reply by raminsalmas** [reply]
Actually when training dataset is used to evaluate the COXPH model, it's able to report the accuracy - but when it comes to random forest method, training dataset shouldn't be used to evaluate the model as it will give a extreme high score, which is not true. I'm wondering if anyone knows the reason of this. 

I think in the linear models which have coefficients it doesn't matter that training dataset is used in the evaluation of the model but in the ensemble models it does.

**Reply by sebp** [reply]
That's no surprising. The Random Survival Forest is a much more complex model than CoxPH and thus can overfit on the training data more easily. Since CoxPH is just a linear model, it doesn't have the capability to overfit on the trainin data.
